extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -1000,4 +991,74 @@ private boolean possiblyPause(Set<Integer> assignment) throws InterruptedExcepti
 
     return false;
   }
+
+  private void possiblyResetOffsetsOrWait(
+      Map<TopicPartition, Long> outOfRangePartitions,
+      KafkaConsumer<byte[], byte[]> consumer,
+      Set<Integer> assignment
+  ) throws InterruptedException
+  {
+    boolean shouldRetry = false;
+    for (Map.Entry<TopicPartition, Long> outOfRangePartition : outOfRangePartitions.entrySet()) {
```

Comment:
why do we need to run this code when `resetOffsetAutomatically ` is set to false.